### ISSUE TITLE
Customization

### ISSUE THREAD

**Issue by schani** [quicktype/quicktype#37]
For the backends we can start with boolean options, for all languages we'd like the ability to at least give a top level class name.

We can later expand this to typed options.  It seems that [generics](https://pursuit.purescript.org/packages/purescript-generics/4.0.0/docs/Data.Generic) can be used for this.

**Comment by dvdsgl** [comment]
How about something like this:

```purescript
module Options where

foreign import data Options :: Type

foreign import optionsEmpty :: Options
foreign import optionsAdd :: forall a. Options -> Option a -> Options
foreign import optionsGet :: forall a. Options -> String -> Maybe (Option a)

data OptionKind
    = Boolean
    | String
    | Choice (Array String)

type Option a =
    { kind :: OptionKind
    , value :: Json
    , label :: String
    }

booleanOption :: String -> Boolean -> Option Boolean
booleanOption label default =
    { kind: Boolean
    , value: fromBoolean default
    , label
    }

renderProperties :: Option Boolean
renderProperties = booleanOption "Render Properties" true

class OptionValue a where
  getValue :: Option a -> Maybe a
  setValue :: Option a -> a -> Option a

instance optionValueBoolean :: OptionValue Boolean where
  getValue { value } = toBoolean value
  setValue opt b = opt { value = fromBoolean b }
```

Then in `Options.js`:

```js
exports.optionsEmpty = [];

exports.optionsAdd = function (opts) {
  return functions (opt) {
    opts.push(opt);
  };
};

...
```

**Comment by schani** [comment]
Implemented in b91eff833befee2eedc393dc4db956be1bbc3fca
